sql/src/main/java/io/crate/expression/symbol/FuncArg.java

```diff
@@ -46,4 +46,14 @@
      * @return True is casting is possible, false otherwise.
      */
     boolean canBeCasted();
+
+    /**
+     * Checks whether this argument can be converted to the given
+     * DataType without any loss of information.
+     * @param dataType The type to check for lossless conversion.
+     * @return True if lossless conversion is possible, false otherwise.
+     */
+    default boolean isLosslesslyConvertableTo(DataType dataType) {
```

Comment:
This is a bit miss-leading. An `integer` column could be converted lossless to a `long` - yet the current implementation would return `false` and looking at the `Param` code we actually depend on this current implementation to get the behaviour we want.

Reply:
True. We could also embed this directly into `Param` but it would require an `instanceof` check of some sort.

Reply:
Added a fixup which checks directly in `Param`.